armory/scenarios/carla_video_tracking.py

```diff
@@ -27,6 +27,9 @@ def load_test_dataset(self):
     def next(self):
         super().next()
         self.y, self.y_patch_metadata = self.y
+        # Fix for tfdsv4 upgrade - separate batches into list
+        if isinstance(self.y, dict) and "boxes" in self.y:
+            self.y = [{"boxes": batch} for batch in self.y["boxes"]]
```

Comment:
r.e. the issue of tf preprocessing outputting the extra batch dimension: here we take a half-step towards addressing it, by removing the batch dim for `self.y`. We don't touch anything in `self.y_patch_metadata` yet, so the attack will still error out

I see a few options for this PR:
1. Address it for both `y` and `y_patch_metadata` by further modifying scenario code to convert from `(1, N, 4)` to `(N, 4)` within `y_patch_metadata` too
2. Don't address it for either yet (i.e. just do `self.y = [self.y]` in the `if` block)
3. Address it for both `y` and `y_patch_metadata` but instead of modifying scenario code to remove the dimension, we could modify metrics to accept either `(N, 4)` or `(1, N, 4)`. We actually already do this for mAP, see [here](https://github.com/twosixlabs/armory/blob/master/armory/metrics/task.py#L651). For this particular model, we'd also need to overwrite [this](https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/main/art/estimators/object_tracking/pytorch_goturn.py#L625) `predict()` method to remove the batch dimension in `y_init`

The pros/cons of (3) vs (1) are what we need to weigh. For context, this batch dimension issue affects all tracking and detection scenarios, and I don't think we should do a mix of (3) and (1). 

(1) may be the quickest path to get things running, but it feels unusual or unexpected to have scenario code doing further dataset preprocessing. It also may have unexpected side effects in the future when users are subclassing the scenario. If we go (3), though, we're telling users "your model + attack must be compatible with this slightly awkward `(1, N, 4)` format"

Thoughts? There may be a possible option (4) where we apply preprocessing not as a tf map operation but instead after the tf dataset is created (enabling us to do preprocessing as we used to), although this would require some rearchitecting

Reply:
@jprokos26 revisiting this, I think (1) is less bad than (3), bc it at least maintains an interface of sensible shape `(N, 4)` over the awkward `(1, N, 4)`. If we went (3), to reiterate, we're forcing anyone creating an attack or metric to work with the awkward shape.

Happy to discuss this more, but if you agree please update `next()` in `carla_video_tracking.py` to also remove the batch dimension for `y_patch_metadata`

Reply:
Addressed with 4dcd2ca9ff5fd42e1415355a564e10ffa75bb073